custom_components/tuya_local/devices/tuya_000004rime_weather_station.yaml

```diff
@@ -0,0 +1,586 @@
+        name: sensor
+      - id: 101
+        optional: true
+        type: base64
+        name: Payload
+      - id: 101
+        optional: true
+        type: base64
+        name: Year
+        persist: true
+        mapping:
+          - mask: "FF000000000000"
+      - id: 101
+        optional: true
+        type: base64
+        name: Month
+        persist: true
+        mapping:
+          - mask: 00FF0000000000
+          - dps_val: 1
+            value: January
+            mask: 00FF0000000000
+          - dps_val: 2
+            value: February
+            mask: 00FF0000000000
+          - dps_val: 3
+            value: March
+            mask: 00FF0000000000
+          - dps_val: 4
+            value: April
+            mask: 00FF0000000000
+          - dps_val: 5
+            value: May
+            mask: 00FF0000000000
+          - dps_val: 6
+            value: June
+            mask: 00FF0000000000
+          - dps_val: 7
+            value: July
+            mask: 00FF0000000000
+          - dps_val: 8
+            value: August
+            mask: 00FF0000000000
+          - dps_val: 9
+            value: September
+            mask: 00FF0000000000
+          - dps_val: 10
+            value: October
+            mask: 00FF0000000000
+          - dps_val: 11
+            value: November
+            mask: 00FF0000000000
+          - dps_val: 12
+            value: December
+            mask: 00FF0000000000
+      - id: 101
+        optional: true
+        type: base64
+        name: Day
+        persist: true
+        mapping:
+          - mask: "0000FF00000000"
+      - id: 101
+        optional: true
+        type: base64
+        name: Hour
+        persist: true
+        mapping:
+          - mask: "000000FF000000"
+      - id: 101
+        optional: true
+        type: base64
+        name: Minute
+        persist: true
+        mapping:
+          - mask: "00000000FF0000"
+      - id: 101
+        optional: true
+        type: base64
+        name: Second
+        persist: true
+        mapping:
+          - mask: "0000000000FF00"
+      - id: 101
+        optional: true
+        type: base64
+        name: Weekday
+        persist: true
+        mapping:
+          - mask: "000000000000FF"
+          - dps_val: 1
+            value: Monday
+            mask: 00FF0000000000
+          - dps_val: 2
+            value: Tuesday
+            mask: 00FF0000000000
+          - dps_val: 3
+            value: Wednesday
+            mask: 00FF0000000000
+          - dps_val: 4
+            value: Thursday
+            mask: 00FF0000000000
+          - dps_val: 5
+            value: Friday
+            mask: 00FF0000000000
+          - dps_val: 6
+            value: Saturday
+            mask: 00FF0000000000
+          - dps_val: 7
+            value: Sunday
+            mask: 00FF0000000000
+  - entity: sensor
+    category: diagnostic
+    name: Weather
+    dps:
+      - id: 103
+        optional: true
+        type: boolean
```

Comment:
Again there is a conflict here between the boolean and base64 declarations for dp 103.  In this case, only the "Weather icon" attribute seems to provide useful info, so that should be the `sensor` attribute definition, and the raw value (Payload) could remain as an extra attribute for reference.